test/functional/feature_presegwit_node_upgrade.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017-2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test a pre-segwit node upgrading to segwit consensus"""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    softfork_active,
+)
+
+class SegwitUpgradeTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 1
+
+    def setup_network(self):
+        self.setup_nodes()
+
+    def run_test(self):
+        self.test_upgrade_after_activation()
+
+    def test_upgrade_after_activation(self):
+        """A pre-segwit node with insufficiently validated blocks needs to redownload blocks"""
+
+        self.log.info("Testing upgrade behaviour for pre-segwit node to segwit rules")
+        node = self.nodes[0]
+
+        # Node hasn't been used or connected yet
+        assert_equal(node.getblockcount(), 0)
+
+        self.restart_node(0, extra_args=["-segwitheight=10"])
```

Comment:
Why not just start the node with `"-segwitheight=10"` first time, rather than starting and then restarting?

Reply:
Nice! Done.